src/main/java/org/apache/maven/plugins/dependency/tree/JsonDependencyNodeVisitor.java

```diff
@@ -110,8 +119,16 @@ private void writeChildren(int indent, DependencyNode node, StringBuilder sb, Se
 
     @Override
     public boolean endVisit(DependencyNode node) {
-        return true;
+        try {
+            if (node.getParent() == null || node.getParent() == node) {
+                writer.flush();
```

Comment:
possibly here and only here content should be written to the stream; I'm not sure